packages/beacon-node/src/api/impl/node/index.ts

```diff
@@ -66,10 +66,18 @@ export function getNodeApi(
       return {data: sync.getSyncStatus()};
     },
 
-    async getHealth(_req, res) {
+    async getHealth(options, _req, res) {
```

Comment:
Uhmm I didn't noticed that request and response objects where being leaked to the API implementation. I see it was introduced by https://github.com/ChainSafe/lodestar/pull/4994 but why not keep the interaction in the Fastify's response object in `packages/api/src/beacon/server/node.ts`?

Reply:
Looked at this as well but the only "clean way" to achieve this without too many type hacks is approach used in https://github.com/ChainSafe/lodestar/pull/3349 but in this case the `getHealth` type is incorrect, it doesn't actually return a `res.response` of type `number`.

If you think passing response object to API implementation is not desired we should update types accordingly and use the other approach to implementing this.

Reply:
@nazarhussain any opinions here?

Reply:
Passing `res` directly to API handler is a pattern which does not meant it's leaking anything. 

We can define a data structure for the implementation of API endpoints, that way we can handle the status code at the server implementation side. 

So all implementation handlers should return; 
```
{
   body: unknown;
   status?: number; 
}
```

And if `status` is not present then we consider `200` as default.

Reply:
Health API is really the only one right now that needs to modify the status code.

The interface of response is limited to setting the status code
https://github.com/ChainSafe/lodestar/blob/7b5fc63e678c4d9010b8b6993a17a8db84667013/packages/api/src/interfaces.ts#L27

Reply:
In that case we can make the response status code structure as optional. e.g.

```
() => T | {body: T, status; number}
```

This will require no refactor on API implementation side and minimum change on the server side.

Reply:
True, that could be a good pattern to give some API implementations more control over the response. Maybe we want to allow to set `null` explicitly as body as well, that way we can more closely follow the spec for APIs that should not return a json body (see https://github.com/ChainSafe/lodestar/pull/5790#issuecomment-1648412274).

Proposed type would look like this then

```ts
() => T | {body: T | null, status; number}
```

Reply:
So will you make the change in this PR or separate?

Reply:
Better to do it in this PR, allows to remove the workaround in tests, should reduce overall diff

Reply:
Request and response objects are no longer passed to API implementation. After looking into updating return type to something like `() => T | {body: T | null, status: number}` it does not look that clean and requires to many type casts in other place unrelated to health API.

Better to address in a separate PR.